src/pudl/helpers.py

```diff
@@ -1285,6 +1284,22 @@ def fillna_w_rolling_avg(
     return df_new.drop(columns=f"{data_col}_rollfilled")
 
 
+def groupby_agg_label_unique_source_or_mixed(x: pd.Series) -> str | None:
+    """Get either the unique source in a group or return mixed.
+
+    Custom function for groupby.agg. Written specifically for
+    aggregating records with fuel_cost_per_mmbtu_source.
+    """
+    sources = [source for source in x.tolist() if isinstance(source, str)]
+    if len(sources) > 1:
+        source = "mixed"
+    elif len(sources) == 1:
+        source = sources[0]
+    else:
+        source = pd.NA
+    return source
```

Comment:
any time we were previously aggregating the `fuel_cost_from_eiaapi`  - which we did many times to make the frc table monthly or annual - we were just using `any`. This custom gb agg function replaces that functionality but instead of saying True for fuel_cost_from_eiaapi if any record came from the api, now it says "mixed" if there are more than one non-null sources.

Unsurprisingly the annual tables are mostly mixed while the monthly aggregation has a pretty even mix of outputs from this.

Reply:
Can we say anything explicit about the proportion of values in the final output? (original, eiaapi, rolling_avg, and still NULL) or anything about how the source varies by time, location, or energy source code?

Reply:
I just did a quick look at the value_counts of the YS and MS aggregated table.

Monthly:
```python
fuel_cost_per_mmbtu_source
original       97146
mixed          80118
eiaapi         64646
<NA>           28322
rolling_avg     5876
```

Yearly:
```python
fuel_cost_per_mmbtu_source
mixed          23844
<NA>            1865
original         519
eiaapi           419
rolling_avg       62
Name: count, dtype: int64
```

Are you suggesting I add a validation tests about this?

Reply:
I was suggesting a way to visually sanity check or better understand the consequences of all of this filling, since it's affecting a large number of records, and it's a modification that we're responsible for.

Maybe once it's merged in I can make a `pudl-examples` notebook that looks at this.

Reply:
I was going spot visual spot checking of the two known to be bad plants as well as a handful of others over the in issue. we also have a rn very not working wip notebook which i believe does what you are thinking about... I can probably update it quickly. But if it takes a long time that would feel oos for this fix the bug issue.

Reply:
Like I said this is not blocking. If it's any hassle don't worry about it and I can play with it on my own time for my own sanity or as an example notebook under our example notebook documentation budget.

Reply:
okay i updated the notebook and will check in shortly. here are some example peeps at some outputs

![image](https://github.com/user-attachments/assets/6dbfa9c0-9f92-4b60-b643-7d4ccf968098)

some example plants 
![image](https://github.com/user-attachments/assets/4f633eeb-9524-4536-9cd9-cd6e2cdf69d1)
![image](https://github.com/user-attachments/assets/546c997b-b93f-4164-b4a3-cdc84f409338)
![image](https://github.com/user-attachments/assets/4371222f-2de1-4b14-8ecd-98b9740cd1e4)
![image](https://github.com/user-attachments/assets/3570961c-209b-4366-a5b0-ed06fee045ef)

aaaand by fuel type
![image](https://github.com/user-attachments/assets/11d132ef-7f97-42f4-ae37-7e2870a0a4d3)